crypto/evp/evp_lib.c

```diff
@@ -58,8 +47,44 @@ int EVP_CIPHER_param_to_asn1(EVP_CIPHER_CTX *c, ASN1_TYPE *type)
         default:
             ret = EVP_CIPHER_set_asn1_iv(c, type);
         }
-    } else
-        ret = -1;
+    } else {
+        if (cipher->prov != NULL) {
+            /* For provider, we simply pass the DER blob */
+            OSSL_PARAM params[3], *p = params;
+            unsigned char *der = NULL, *derp;
+
+            /*
+             * We do two passes, the first to get the appropriate buffer
+             * size, and the second to get the actual value.
+             */
+            *p++ = OSSL_PARAM_construct_octet_string(OSSL_CIPHER_PARAM_ALG_ID,
+                                                     NULL, 0);
+            *p = OSSL_PARAM_construct_end();
+
+            /* No point checking, we know we get an error. */
+            EVP_CIPHER_CTX_get_params(c, params);
```

Comment:
I dont think it does get an error?

Reply:
Hmm, I think the OSSL_PARAM functions do... which might actually be faulty. I will have to check.

Reply:
Anyway, I changed this to check for errors.